scripts/west_commands/boards.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) 2019 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import collections
+import os
+import re
+import shutil
+import subprocess
+import sys
+import textwrap
+
+from west import log
+from west import cmake
+from west.configuration import config
+from west.commands import WestCommand
+from cmake import run_cmake
+
+BOARDS_DESCRIPTION = '''\
+Display information about boards
+'''
+
+class Boards(WestCommand):
+
+    def __init__(self):
+        super(Boards, self).__init__(
+            'boards',
+            # Keep this in sync with the string in west-commands.yml.
+            'display information about supported boards',
+            BOARDS_DESCRIPTION,
+            accepts_unknown_args=False)
+
+    def do_add_parser(self, parser_adder):
+        default_fmt = '{name:32} ({arch})'
+        parser = parser_adder.add_parser(
+            self.name,
+            help=self.help,
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            description=self.description,
+            epilog=textwrap.dedent('''\
+            FORMAT STRINGS
+
+            Boards are listed using a Python 3 format string. Arguments
+            to the format string are accessed by name.
+
+            The default format string is:
+
+            "{}"
+
+            The following arguments are available:
+
+            - name: board name
+            - arch: board architecture
+            '''.format(default_fmt)))
+
+        # Remember to update scripts/west-completion.bash if you add or remove
+        # flags
+        parser.add_argument('-f', '--format', default=default_fmt,
+                 help='''Format string to use to list each board; see
+                 FORMAT STRINGS below.'''),
+
+        return parser
+
+    def do_run(self, args, unknown_args):
+
+        zb = os.environ.get('ZEPHYR_BASE')
+        if not zb:
+            log.die('Internal error: ZEPHYR_BASE not set in the environment, '
+                    'and should have been by the main script')
+
+        cmake_args = ['-DBOARD_ROOT_SPACE_SEPARATED={}'.format(zb),
+                      '-P', '{}/cmake/boards.cmake'.format(zb)]
+        lines = run_cmake(cmake_args, capture_output=True)
+        arch_re = re.compile(r'\s*([\w-]+)\:')
+        board_re = re.compile(r'\s*([\w-]+)\s*')
+        arch = None
+        boards = collections.OrderedDict()
+        for line in lines:
+            match = arch_re.match(line)
```

Comment:
Just as a tip in case you are calling `re.compile` out of performance concerns: the `re` module maintains a cache of recently used regular expressions, so since only two are being used here, there shouldn't be any speedup from calling compile manually.

Reply:
It's mostly habit actually. I have always used Python's regex module like this. I guess it's about the same either way in terms of performance so I won't change this.

Reply:
Thanks for the tip though! :)